File: frontend/lib/data/models/question.dart
enum QuestionType { SELECT, ENTER }

class Question {
  final String questionID;
  final String title;
  final QuestionType type;
  final List<String> options;
  final String correct;

  Question({
    required this.questionID,
    required this.title,
    required this.type,
    required this.options,
    required this.correct,
  });
}

File: backend/src/main/java/com/app/mevocab/internal/quiz/dto/Question.java
package com.app.mevocab.internal.quiz.dto;

import com.app.mevocab.common.enums.QuestionType;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.util.HashSet;
import java.util.List;
@Builder(toBuilder = true)
@Data
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Question {
    Integer questionID;
    String title;
    QuestionType type;
    List<String> options;
    String correct;

    public boolean checkQuestion() {
        if (type.equals(QuestionType.ENTER)) {
            int count = title.split("=@=", -1).length - 1;
            return count == 1;
        } else {
            return options != null && options.contains(correct);
        }
    }
}
